argus/src/main/java/com/moldedbits/argus/handler/BaseProviderHandler.java

```diff
@@ -0,0 +1,48 @@
+package com.moldedbits.argus.handler;
+
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.moldedbits.argus.ArgusTheme;
+import com.moldedbits.argus.R;
+import com.moldedbits.argus.provider.BaseProvider;
+import com.moldedbits.argus.utils.ThemeHelper;
+
+/*
+    Handler which will :-
+  * Apply Theme to SignIn and SignUp Providers
+  * Provides login functionality for specific end point
+ */
+public abstract class BaseProviderHandler extends BaseProvider implements ThemeHelper {
```

Comment:
The thing was I wanted the application of themes and providers at a single place plus it was not right to have this apply theme in BaseProvider as it is a provider <- I'm not sure of this though